File: plant_twin/lib/models/reminder.dart
class Reminder {
  final int id;
  final int plantId;
  final String type;
  final String nextDueDate;
  final String frequency;
  final bool isCompleted;

  Reminder({
    required this.id,
    required this.plantId,
    required this.type,
    required this.nextDueDate,
    required this.frequency,
    required this.isCompleted,
  });

  factory Reminder.fromJson(Map<String, dynamic> json) {
    return Reminder(
      id: json['id'],
      plantId: json['plant_id'],
      type: json['reminder_type'], // Matches Pydantic schema
      nextDueDate: json['next_due_date'],
      frequency: json['frequency'],
      isCompleted: json['is_completed'],
    );
  }
}

File: backend/app/models/reminder.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from app.database import Base

class Reminder(Base):
    __tablename__ = "reminders"
    id = Column(Integer, primary_key=True, index=True)
    plant_id = Column(Integer, ForeignKey("plants.id"))
    reminder_type = Column(String) # e.g., "water", "fertilizer"
    next_due_date = Column(DateTime)
    frequency = Column(String) # e.g., "daily", "weekly"
    is_completed = Column(Boolean, default=False)

    plant = relationship("app.models.plant.Plant", back_populates="reminders")
